static/js/settings_config.ts

```diff
@@ -414,7 +425,15 @@ export const all_notification_settings = other_notification_settings.concat(
     stream_notification_settings,
 );
 
-export function get_notifications_table_row_data(notify_settings) {
+interface NotificationSetting {
```

Comment:
Maybe NotificationSettingCheckbox, since these are all booleans?

Reply:
Also, would it make sense to declare this as an actual type, rather than an `interface`, and then create objects of that type in the functions below?

Reply:
Renamed and changed to use a type (`type NotificationSettingCheckbox = ...`). In this case, it doesn't matter if we use `type` or `interface` they both do the same thing.

Reply:
Here is the [typescript docs](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces) that explains the difference between the two.

Reply:
Yeah, I think Type seems better for this.